### PR TITLE
RetryPolicy Proposal

### PR DESCRIPTION
**Pull request by alex268**

### CODE REVIEW

**Review by KirillKurdyukov — APPROVED** [review]

File: core/src/main/java/tech/ydb/core/RetryPolicy.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Abstracts the policy to use when retrying some actions
+ *
+ * @author Aleksandr Gorshenin
+ */
+public interface RetryPolicy {
+    /**
+     * Called when an operation has failed for some reason. This method may return next values
+     * <ul>
```
Comment (pnv1) [inline core/src/main/java/tech/ydb/core/RetryPolicy.java:11]: This is better be moved to @return section IMHO
The title could be extended to describe the purpose of this method. Something like this:
"Called when an operation is failed for some reason to determine if it should be retried. And if so, returns the delay to make the next retry attempt after"

File: core/src/main/java/tech/ydb/core/retry/MaxElapsedRetry.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package tech.ydb.core.retry;
+
+/**
+ *
+ * @author Aleksandr Gorshenin
+ */
+public class MaxElapsedRetry extends ExponentialBackoffRetry {
```
Comment (pnv1) [inline core/src/main/java/tech/ydb/core/retry/MaxElapsedRetry.java:7]: MaxElapsedRetryPolicy?

File: core/src/main/java/tech/ydb/core/retry/ForeverRetry.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package tech.ydb.core.retry;
+
+import tech.ydb.core.RetryPolicy;
+
+/**
+ *
+ * @author Aleksandr Gorshenin
+ */
+public class ForeverRetry implements RetryPolicy {
```
Comment (pnv1) [inline core/src/main/java/tech/ydb/core/retry/ForeverRetry.java:9]: InfiniteRetryPolicy?

File: core/src/main/java/tech/ydb/core/retry/ExponentialBackoffRetry.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.concurrent.ThreadLocalRandom;
+
+import tech.ydb.core.RetryPolicy;
+
+/**
+ *
+ * @author Aleksandr Gorshenin
+ */
+public abstract class ExponentialBackoffRetry implements RetryPolicy {
```
Comment (pnv1) [inline core/src/main/java/tech/ydb/core/retry/ExponentialBackoffRetry.java:11]: ExponentialBackoffRetryPolicy?

**Review by pnv1 — APPROVED** [review]

**Pull request merged**